examples/pxScene2d/src/pxContextGL.cpp

```diff
@@ -121,6 +121,7 @@ extern rtScript script;
 extern uv_async_t gcTrigger;
 #endif
 extern pxContext context;
+rtMutex glContextLock;
```

Comment:
Currently, the pxContext APIs are called from the same thread (in our standard case, this is the main thread).  Can you please review the source and see if you can call pxContext APIs on the same thread and remove the need for the glContextLock mutex?

Reply:
Yea, also as per your comments I added support in the WPE Spark Plugin to make sharedContext for suspend/resume action, since it is happening from the Plugin Context. https://github.com/WebPlatformForEmbedded/WPEFrameworkPlugins/blob/master/Spark/SparkImplementation.cpp#L300. 
But after adding this change, I am getting random deadlock due to thread_mutex_lock because of parallel call of makeCurrent and texture->bindGLTexture(). I have added this lock to avoid this issue.